soundfile.py

```diff
@@ -986,7 +986,9 @@ def _open(self, file, mode_int, closefd):
             file_ptr = _snd.sf_open(file, mode_int, self._info)
         elif isinstance(file, int):
             file_ptr = _snd.sf_open_fd(file, mode_int, self._info, closefd)
-        elif all(hasattr(file, a) for a in ('seek', 'read', 'write', 'tell')):
+        elif ( hasattr(file, 'seek') and hasattr(file, 'tell') and
```

Comment:
> his improves readability

It doesn't for me, but I guess this is highly subjective.

For me it just looks awkward and French.

> I will not take pep8 as gospel.

Me neither, but I think it's a good tie-breaker ...

Consistency is a great thing, PEP8 merely helps us reaching it.
There may be reasons to allow inconsistencies, but in the current case I don't see any upside.

Probably factoring out the if-condition into a proper function would help?

BTW, adding a condition for `'readinto'` will only make this worse ...